File: 코드리뷰/장문주(jmj2359)/6일차/코드리뷰_밧줄.py
import sys
n = int(sys.stdin.readline())
sum = 0
l = []
for  i in range(n):
    l.append(int(sys.stdin.readline()))

l.sort(reverse=True)
m = 0
c = 1
t = 0

for i in l:
    if t < i*c:
        t = i*c
    if i*c >= m:
        c+=1
        sum+=i
    m = sum/(c-1)
print(t)

File: 코드리뷰/최시운(siun0331)/6일차/코드리뷰_밧줄.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

public class main {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        ArrayList<Integer> list = new ArrayList<>();

        int n = Integer.parseInt(br.readLine());
        for(int i = 0; i<n; i++){
            int x = Integer.parseInt(br.readLine());
            list.add(x);
        }

        Collections.sort(list, Comparator.reverseOrder());

        int index = list.size();
        int max = 0;

        for(int i = 0; i < n; i++){
            int sum = 0;
            for(int j = 0; j < index; j++){
                sum = list.get(index-1)*index;
            }
            if (sum > max) max = sum;
            index --;
        }

        System.out.println(max);

    }
}